steamclog/src/main/java/com/steamclock/steamclog/Destinations.kt

```diff
@@ -186,38 +204,44 @@ internal class ExternalLogFileDestination : Timber.DebugTree() {
         return logDirectory
     }
 
-    fun getExternalFile(): File? {
+    /**
+     * Will get the external file we are currently writing to.
+     */
+    private fun getExternalFile(): File? {
         val expiryMs = SteamcLog.config.autoRotateConfig.fileRotationSeconds * 1000 // defaults to 10 minutes
 
         if (isCachedLogFileValid(expiryMs)) {
             return cachedCurrentLogFile
         }
 
-        val currentFile = findAvailableLogFile(expiryMs)
-        updateCachedLogFile(currentFile)
-        return currentFile
+        // Find new available log file and update cached variables
+        cachedCurrentLogFile = findAvailableLogFile(expiryMs)
+        logFileCreatedTimeEstimate = Date().time
+        return cachedCurrentLogFile
     }
 
     private fun isCachedLogFileValid(expiryMs: Long): Boolean {
         // if you have a cached log file and you checked it within the expiry window
         val now = Date().time
-        val currentLogFileCachedTime = currentLogFileCachedTime
-        return cachedCurrentLogFile != null &&
+        val currentLogFileCachedTime = logFileCreatedTimeEstimate
+        return logFileCreatedTimeEstimate != null &&
                 currentLogFileCachedTime != null &&
```

Comment:
not sure I understand the logic behind this, you just assigned `currentLogFileCachedTime = logFileCreatedTimeEstimate`, so aren't we checking the same thing twice?

Reply:
Ah, I see.
What about doing it this way:
```
val currentLogFileCachedTime = logFileCreatedTimeEstimate ?: return false
```

I think that should ensure  both the file and the estimated start time have been set, and then you should be able to remove both null checks

Reply:
Ah that's a good idea. Makes it much cleaner! Done :)